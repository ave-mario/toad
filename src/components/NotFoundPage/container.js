import { connect } from 'react-redux';
import { push } from 'connected-react-router';
import NotFoundPage from './view';

const mapStateToProps = () => ({});

const mapDispatchToProps = dispatch => {
  return {
    goBack: () => {
      dispatch(push('/login'));
    }
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(NotFoundPage);
