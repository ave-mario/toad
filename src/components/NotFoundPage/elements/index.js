import styled from 'styled-components';
import { Button } from '../../../elements';

const darkGrey = '#4d4d4d';

export const NotFound = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
  max-width: 100vw;
  min-height: 100vh;
`;
export const NotFoundButton = styled(Button)`
  border-radius: 2px;
  margin: 5px auto;
  max-width: 150px;
  max-height: 25px;
  background-color: ${darkGrey};
  transition: background-color 0.3s;
  :hover {
    background-color: #27ae60;
  }
`;
