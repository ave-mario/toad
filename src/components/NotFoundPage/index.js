import NotFoundPageContainer from './container';

export default NotFoundPageContainer;
