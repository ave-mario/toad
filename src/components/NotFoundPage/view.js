import React from 'react';
import PropTypes from 'prop-types';
import { Title, Title3 } from 'elements';
import { NotFound, NotFoundButton } from './elements';

function NotFoundPage(props) {
  const { goBack } = props;
  return (
    <>
      <NotFound>
        <Title3>OOPS! PAGE NOT FOUND</Title3>
        <Title fontSize="5em">404</Title>
        <Title3>WE ARE SORRY, BUT THE PAGE YOU REQUESTED WAS NOT FOUND</Title3>
        <NotFoundButton
          onClick={() => {
            goBack();
          }}
          className="goBackButton"
        >
          GO MAIN PAGE
        </NotFoundButton>
      </NotFound>
    </>
  );
}

NotFoundPage.propTypes = {
  goBack: PropTypes.func.isRequired
};

export default NotFoundPage;
