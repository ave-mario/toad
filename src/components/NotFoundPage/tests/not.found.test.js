import React from 'react';
import { shallow, mount } from 'enzyme';
import NotFoundPage from '../view';

describe('Test case for testing NotFound page', () => {
  test('NotFound page snapshot', () => {
    const props = {
      goBack: () => {}
    };
    const component = shallow(<NotFoundPage {...props} />);

    expect(component).toMatchSnapshot();
  });
  test('should call go back on button click', () => {
    const mockGoBack = jest.fn();
    const props = {
      goBack: mockGoBack
    };
    const component = mount(<NotFoundPage {...props} />);

    component.find('button.goBackButton').simulate('click');
    expect(mockGoBack).toHaveBeenCalled();
  });
});
