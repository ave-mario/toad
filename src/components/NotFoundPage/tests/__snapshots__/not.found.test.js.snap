// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test case for testing NotFound page NotFound page snapshot 1`] = `
<Fragment>
  <styled.div>
    <styled.h3>
      OOPS! PAGE NOT FOUND
    </styled.h3>
    <styled.h1
      fontSize="5em"
    >
      404
    </styled.h1>
    <styled.h3>
      WE ARE SORRY, BUT THE PAGE YOU REQUESTED WAS NOT FOUND
    </styled.h3>
    <Styled(styled.button)
      className="goBackButton"
      onClick={[Function]}
    >
      GO BACK
    </Styled(styled.button)>
  </styled.div>
</Fragment>
`;
